### PR TITLE
Add toggleClass option for conditional to allow greater flexibility

### PR DESCRIPTION
**Pull request by balbuf**
Keep show/hide for BC, but this should have been done with classes from the get-go.

### CODE REVIEW

File: inc/class-wp-forms-api.php [hunk 1]
```diff
@@ -491,7 +491,7 @@ static function render_element( $element, &$values, $form = null ) {
 			}
 
 			// Conditional logic
-			if( $element['#conditional']['element'] && $element['#conditional']['action'] && $element['#conditional']['value'] ) {
+			if( isset( $element['#conditional']['element'], $element['#conditional']['action'], $element['#conditional']['value'] ) ) {
```
Comment (bendoh) [inline inc/class-wp-forms-api.php:497]: TIL you can pass multiple arguments to `isset()`

**Comment by balbuf** [comment]
Yeah I'm just kind of fitting this into the existing structure - it would be nice to revisit this though and make it a tad more flexible, e.g. multiple values to trigger different actions.

File: inc/class-wp-forms-api.php (hunk 1 above)
Comment (balbuf) [inline inc/class-wp-forms-api.php:497]: I wish the same could be said for `empty()` :disappointed:

**Comment by bendoh** [comment]
Well it's /approved, noting that there's a silent failure when `data-conditional-action` has an unmatched value, but I suppose it was like that before.

File: inc/wp-forms-api.js [hunk 2]
```diff
@@ -314,39 +314,29 @@
...
-				$target.hide();
-			}
-			else {
-				$target.show();
-			}
+		if (action === 'show') {
+			$target.toggle(value == inputValue);
+		} else if (action === 'hide') {
+			$target.toggle(value != inputValue);
+		} else if ((action=action.split(' ')) && action[0]==='toggleClass') {
```
Comment (bendoh) [inline inc/wp-forms-api.js]: Please fix the spacing around the operators in this line!

**Comment by jcowher** [comment]
I might I still have my code from my Marketpress days... I spent a solid couple of months on a similar form API that included conditional logic, repeatable fields, etc. I'll see if I can dig it up.

**Comment by balbuf** [comment]
What are y'all's thoughts on how I've restructured it?

**Comment by balbuf** [comment]
@jcowher you could not write it like that because `is()` returns a boolean

**Comment by balbuf** [comment]
@jcowher `conditions = $this.data('conditional') || {}` is not necessary because `undefined` is not an object, and it bails if `conditions` is not an object.

**Comment by balbuf** [comment]
Besides, `:checkbox` is a `jQuery` pseudo selector, so it is already going to use non-standard selector approach.

**Comment by jcowher** [comment]
> Besides, :checkbox is a jQuery pseudo selector, so it is already going to use non-standard selector approach.

Yea, not a huge deal. I, personally, just try to avoid using crazy expressions when ever possible and stick with method chaining because it's usually more efficient. But, for the sake of forest and not trees let's let is slide. :)

**Comment by jcowher** [comment]
> conditions = $this.data('conditional') || {} is not necessary because undefined is not an object, and it bails if conditions is not an object.

Yea, that's true. I always get tripped up with undefined. Depending on how you're using the undefined var, you could get errors. For example: I think `if (var)` is good, but `if (var == 'poop')` would throw an error.

**Comment by balbuf** [comment]
Well `var` cannot be a variable name :) but `if (foo == 'poop')` would not throw an error if `foo` has the value undefined, only if the variable `foo` was never declared. `undefined` is a perfectly legal value!

**Comment by balbuf** [comment]
And I think what you are saying regarding selectors is a valid point, but considering `:checkbox` is a jQuery-ism, it's going to have to rely on sizzle regardless. This seems to be an issue of micro-optimization IMO.

**Comment by jcowher** [comment]
Yea, I think we're good. Great discussion though! /approved :mega:

**Comment by balbuf** [comment]
Hey y'all...another little update to fix the handling of radio elements.

**Comment by balbuf** [comment]
All good?

**Comment by jcowher** [comment]
Yup! /approved

**Pull request merged**